Serializer/JsonLdNormalizer.php

```diff
@@ -258,12 +258,25 @@ public function denormalize($data, $class, $format = null, array $context = [])
                     if ($attributes[$attribute]->getTypes()[0]->isCollection()) {
                         $collection = [];
                         foreach ($value as $uri) {
+                            if (!is_string($uri)) {
+                                throw new InvalidArgumentException(sprintf(
+                                    'Nested object are not supported (found in attribute "%s")',
+                                    $attribute
+                                ));
+                            }
+
                             $collection[] = $this->dataManipulator->getObjectFromUri($uri);
                         }
 
                         $value = $collection;
-                    } else {
+                    } else if (is_string($value)) {
```

Comment:
Nop, it's necessary else the `Get an error during deserialization of simple relation` scenario will fail.